### DISCUSSION TITLE
how to create little rect  in a big rect

### DISCUSSION THREAD

**Discussion by wobushishuaige222** [fabricjs/fabric.js#7304] [Q&A]
Hi guys,
I need to create another rect in a rect, but when I click on the outside rect, it is selected by default, and I cannot create another rect  in this rect.I had thought about use "selectable" property to remove its event. But it too hard to contriol,so do u have the useful property ， API  or some good ideas,
such like the bottom example
![rectInrect](https://user-images.githubusercontent.com/38483556/129297045-f52d617f-4da8-4ef4-a012-4bbb4d7df259.png)

**Comment by gloriousjob** [comment]
One idea: If you have a trigger, like a button to create rectangles, you could have the button make all current objects not selectable when clicked. After the rectangle is drawn, all objects, including the rectangle just drawn, could have the selectable flag set again.

**Comment by wobushishuaige222** [comment]
> 一个想法：如果你有一个触发器，比如创建矩形的按钮，你可以让按钮在点击时使所有当前对象不可选。绘制矩形后，所有对象，包括刚刚绘制的矩形，都可以再次设置可选标志。



> One idea: If you have a trigger, like a button to create rectangles, you could have the button make all current objects not selectable when clicked. After the rectangle is drawn, all objects, including the rectangle just drawn, could have the selectable flag set again.



> One idea: If you have a trigger, like a button to create rectangles, you could have the button make all current objects not selectable when clicked. After the rectangle is drawn, all objects, including the rectangle just drawn, could have the selectable flag set again.



> One idea: If you have a trigger, like a button to create rectangles, you could have the button make all current objects not selectable when clicked. After the rectangle is drawn, all objects, including the rectangle just drawn, could have the selectable flag set again.

I am still not sure what you are talking about? Could u tell me more detail information? My business requirement is that the way to create rectangles can only be dragged by the mouse.

**Comment by gloriousjob** [comment]
Like this: http://fabricjs.com/freedrawing

Instead of a free drawing button, you have a draw rectangles button or 
whatever your need is.

**Comment by wobushishuaige222** [comment]
It seem  like a good idea.But my product manager dont agree because of the user experience.Thank very much for your reply

**Comment by asturur** [comment]
There are so many un-optimal way here to ask for help.
Creating 2 rects inside one another is a so generic need that no one can really give you an answer that works.
There is no context.
Is the user creating it? do you create them with code? is there a button? is a mouse gesture?
Please open an issue just when you have some working code that behave strange and you want to understand if there is a problem.

**Comment by dmdum** [comment]
Hey @wobushishuaige222, I'm trying to do something similar, were you able to find a solution for this?
